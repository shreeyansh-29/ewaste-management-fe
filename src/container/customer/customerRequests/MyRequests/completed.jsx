/* eslint-disable indent */
import React, {useEffect} from "react";
import MaterialTable from "material-table";
import Popup from "../../../../components/popup";
import "../../customer.css";
import {
  ProfileIconBarStyle,
  ProfileIconStyle,
} from "../../../../components/styles";
import {FaUserCircle} from "react-icons/fa";
import SearchIcon from "@material-ui/icons/Search";
import {toast} from "react-toastify";
export const ProfileIcon = FaUserCircle;
import {useDispatch, useSelector} from "react-redux";
import {customerCompletedRequest} from "../../../../redux/action/customer/customerCompletedRequestAction/customerCompletedRequestAction";
import {isEmpty} from "lodash";
toast.configure();

export default function completed() {
  const dispatch = useDispatch();
  let res = useSelector((state) => state.customerCompletedRequest);
  const {useState} = React;
  const [isopen, setopen] = useState(false);
  const [detail, setdetail] = useState();

  const [columns] = useState([
    {
      title: "Request Name",
      field: "itemName",
      editable: "never",
      cellStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
    },
    {
      title: "Category",
      field: "category",
      editable: "never",
      lookup: {
        Temp: "Temperature exchange equipment ",
        Screens: "Screens, monitors ",
        Lapms: "Lamps ",
        LargeEqip: "Large equipment",
        SmallEquip: "Small equipment ",
        SmallIT:
          "Small IT and telecommunication equipment (such as mobile phones, printers)",
      },
      cellStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
    },
    {
      title: "Quantity",
      field: "quantity",
      editable: "never",
      cellStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
    },
    {
      title: "Request Type",
      field: "requestType",
      editable: "never",
      cellStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
    },
    {
      title: "Date",
      field: "scheduledDate",
      editable: "never",
      cellStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
    },
    {
      title: "Time",
      field: "scheduledTime",
      editable: "never",
      cellStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
    },

    {
      title: "Status",
      field: "status",
      editable: "never",
      cellStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
    },
  ]);

  const togglepop = () => {
    setopen(!isopen);
  };
  const handledata = (res1) => {
    res1.map((obj) => {
      if (obj.scheduledDate !== null) {
        const date = obj.scheduledDate.split("T");
        obj.scheduledDate = date[0];
      }
    });
  };
  useEffect(() => {
    if (isEmpty(res?.data) !== true) {
      handledata(res?.data.data);
    }
  });
  useEffect(() => {
    dispatch(customerCompletedRequest());
  }, []);

  return (
    <>
      <div>
        <div>
          <div style={{padding: " 10px 30px"}}>
            <MaterialTable
              align="center"
              title=""
              columns={columns}
              data={res?.data.data}
              icons={{
                Search: () => <SearchIcon style={{fill: "white"}} />,
              }}
              localization={{
                header: {
                  actions: "Actions",
                },
              }}
              actions={[
                {
                  icon: () => (
                    <>
                      <ProfileIconStyle onClick={togglepop}>
                        <ProfileIconBarStyle>
                          <ProfileIcon></ProfileIcon>
                        </ProfileIconBarStyle>
                      </ProfileIconStyle>
                    </>
                  ),

                  onClick: (e, datas) => {
                    e.preventDefault();
                    setdetail(datas.collectorUid);
                  },
                },
              ]}
              options={{
                actionsColumnIndex: -1,
              }}
            />
          </div>
        </div>
        <div>
          {isopen && detail != null && (
            <Popup handleClose={togglepop} contents={detail} />
          )}
        </div>
      </div>
    </>
  );
}
