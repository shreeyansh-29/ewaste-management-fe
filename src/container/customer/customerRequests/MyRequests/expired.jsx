import React, {useEffect} from "react";
import MaterialTable from "material-table";
import "../../customer.css";
import SearchIcon from "@material-ui/icons/Search";
import {toast} from "react-toastify";
import {customerExpiredRequest} from "../../../../redux/action/customer/customerExpiredRequestAction/customerExpiredRequestAction";
import {useDispatch, useSelector} from "react-redux";
toast.configure();

const Completed = () => {
  const {useState} = React;
  const dispatch = useDispatch();
  let res = useSelector((state) => state.customerExpiredRequest?.data);

  const [columns] = useState([
    {
      title: "ID",

      field: "id",
      editable: "never",
      cellStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
    },
    {
      title: "Request Name",
      field: "itemName",
      editable: "never",
      cellStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
    },
    {
      title: "Category",
      field: "category",
      editable: "never",
      lookup: {
        Temp: "Temperature exchange equipment ",
        Screens: "Screens, monitors ",
        Lapms: "Lamps ",
        LargeEqip: "Large equipment",
        SmallEquip: "Small equipment ",
        SmallIT:
          "Small IT and telecommunication equipment (such as mobile phones, printers)",
      },
      cellStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
    },
    {
      title: "Quantity",
      field: "quantity",
      editable: "never",
      cellStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
    },
    {
      title: "Request Type",
      field: "requestType",
      editable: "never",
      cellStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
    },
    {
      title: "Date",
      field: "scheduledDate",
      editable: "never",
      cellStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
    },
    {
      title: "Time",
      field: "scheduledTime",
      editable: "never",
      cellStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
    },
  ]);

  useEffect(() => {
    dispatch(customerExpiredRequest());
  }, []);

  return (
    <div>
      <div>
        <div style={{padding: " 10px 30px"}}>
          <MaterialTable
            align="center"
            title=""
            columns={columns}
            data={res?.data}
            icons={{
              Search: () => <SearchIcon style={{fill: "white"}} />,
            }}
            options={{
              actionsColumnIndex: -1,
            }}
          />
        </div>
      </div>
    </div>
  );
};
export default Completed;
