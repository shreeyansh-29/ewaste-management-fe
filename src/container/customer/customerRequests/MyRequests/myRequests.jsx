import React, {useState} from "react";
import Tabs from "@mui/material/Tabs";
import Tab from "@mui/material/Tab";
import Paper from "@mui/material/Paper";
import Typography from "@material-ui/core/Typography";
import Box from "@material-ui/core/Box";
import Pending from "./pending";
import Completed from "./completed";

const MyRequests = () => {
  const [value, setValue] = useState(0);
  const handleChange = (event, newValue) => {
    setValue(newValue);
  };
  const TabPanel = (props) => {
    const {children, index, ...other} = props;
    return (
      <div
        role="tabpanel"
        hidden={value !== index}
        id={`simple-tabpanel-${index}`}
        aria-labelledby={`simple-tab-${index}`}
        {...other}
      >
        {value === index && (
          <Box>
            <Typography>{children}</Typography>
          </Box>
        )}
      </div>
    );
  };
  return (
    <div style={{padding: "150px 30px 0 30px"}}>
      <h2
        style={{
          textAlign: "center",
          fontSize: "30px",
          padding: "2px,",
          color: "white",
          marginBottom: "2.5%",
          backgroundColor: " rgb(30, 28, 54)",
          borderRadius: "5px",
        }}
      >
        {" "}
        My Requests{" "}
      </h2>
      <Paper square style={{margin: "20px auto", width: "100%"}}>
        <Tabs
          value={value}
          indicatorColor="primary"
          textColor="primary"
          onChange={handleChange}
          aria-label="disabled tabs example"
          centered
          variant="fullWidth"
          elevation={20}
        >
          <Tab label="Pending" />

          <Tab label="Completed" />
        </Tabs>
        <TabPanel value={value} index={0}>
          <Pending />
        </TabPanel>
        <TabPanel value={value} index={1}>
          <Completed />
        </TabPanel>
      </Paper>
    </div>
  );
};
export default MyRequests;
