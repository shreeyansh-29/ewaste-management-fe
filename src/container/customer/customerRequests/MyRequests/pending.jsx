import React, {useEffect} from "react";
import MaterialTable from "material-table";
import "../../customer.css";
import {FaUserCircle} from "react-icons/fa";
import api from "../../../../core/utilities/httpProvider";
import SearchIcon from "@material-ui/icons/Search";
import {toast} from "react-toastify";
import {CUSTOMER_REQUEST_PENDING} from "../../../constant/constant";
export const ProfileIcon = FaUserCircle;
toast.configure();

const Pending = () => {
  const {useState} = React;

  const [columns] = useState([
    {
      title: "ID",

      field: "id",
      editable: "never",
      cellStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
    },
    {
      title: "Request Name",
      field: "itemName",
      editable: "never",
      cellStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
    },
    {
      title: "Category",
      field: "category",
      editable: "never",
      lookup: {
        Temp: "Temperature exchange equipment ",
        Screens: "Screens, monitors ",
        Lapms: "Lamps ",
        LargeEqip: "Large equipment",
        SmallEquip: "Small equipment ",
        SmallIT:
          "Small IT and telecommunication equipment (such as mobile phones, printers)",
      },
      cellStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
    },
    {
      title: "Quantity",
      field: "quantity",
      editable: "never",
      cellStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
    },
    {
      title: "Request Type",
      field: "requestType",
      editable: "never",
      cellStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
    },
    {
      title: "Date",
      field: "scheduledDate",
      editable: "never",
      cellStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
    },
    {
      title: "Time",
      field: "scheduledTime",
      editable: "never",
      cellStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "13px",
      },
    },
  ]);

  const handledate = (res) => {
    res.data.map((obj) => {
      if (obj.requestType === "PickUp") {
        if (obj.scheduledTime === "10") {
          obj.scheduledTime = " 10:00-12:00";
        } else if (obj.scheduledTime === "12") {
          obj.scheduledTime = " 12:00-14:00";
        } else if (obj.scheduledTime === "14") {
          obj.scheduledTime = " 14:00-16:00";
        } else if (obj.scheduledTime === "16") {
          obj.scheduledTime = " 16:00-18:00";
        }
      }
    });
  };
  const handleDecline = async (e, datas) => {
    console.log(datas);
    const tokens = localStorage.getItem("token");
    const email = localStorage.getItem("email");

    const response = await fetch(
      `http://localhost:8083/customer/deleteById?orderId=${datas.orderUid}`,
      {
        method: "DELETE",
        credentials: "same-origin",
        headers: {
          "Content-Type": "application/json",
          Authorization: "Bearer " + tokens,
          EMAIL: email,
        },
      }
    );
    console.log(response);
    document.location.reload();
  };
  const handledata = (res) => {
    handledate(res);
    res.data.map((obj) => {
      if (obj.scheduledDate !== null) {
        const date = obj.scheduledDate.split("T");
        obj.scheduledDate = date[0];
      }
      if (obj.requestType === "PickUp") {
        obj.id = "CP" + obj.id;
      }
      if (obj.requestType === "DropOff") {
        obj.id = "CD" + obj.id;
      }
    });
  };
  useEffect(() => {
    (async function () {
      const res = await api.get(CUSTOMER_REQUEST_PENDING);
      console.log(res);
      if (res.status === "success") {
        handledata(res);
        setData(res.data);
      }
    })();
  }, []);
  const [data, setData] = useState();

  return (
    <div>
      <div>
        <div style={{padding: " 10px 30px"}}>
          <MaterialTable
            align="center"
            title=""
            columns={columns}
            data={data}
            icons={{
              Search: () => <SearchIcon style={{fill: "white"}} />,
            }}
            localization={{
              header: {
                actions: "Actions",
              },
            }}
            actions={[
              {
                icon: () => <button className="bttn"> Decline </button>,
                onClick: (e, datas) => handleDecline(e, datas),
              },
            ]}
            options={{
              actionsColumnIndex: -1,
            }}
          />
        </div>
      </div>
    </div>
  );
};
export default Pending;
