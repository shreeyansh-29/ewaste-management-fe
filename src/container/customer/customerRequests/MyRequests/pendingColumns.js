export const pendingColumns = [
  {
    title: "ID",

    field: "id",
    editable: "never",
    cellStyle: {
      textAlign: "center",
      fontSize: "13px",
    },
    headerStyle: {
      textAlign: "center",
      fontSize: "13px",
    },
  },
  {
    title: "Request Name",
    field: "itemName",
    editable: "never",
    cellStyle: {
      textAlign: "center",
      fontSize: "13px",
    },
    headerStyle: {
      textAlign: "center",
      fontSize: "13px",
    },
  },
  {
    title: "Category",
    field: "category",
    editable: "never",
    lookup: {
      Temp: "Temperature exchange equipment ",
      Screens: "Screens, monitors ",
      Lapms: "Lamps ",
      LargeEqip: "Large equipment",
      SmallEquip: "Small equipment ",
      SmallIT:
        "Small IT and telecommunication equipment (such as mobile phones, printers)",
    },
    cellStyle: {
      textAlign: "center",
      fontSize: "13px",
    },
    headerStyle: {
      textAlign: "center",
      fontSize: "13px",
    },
  },
  {
    title: "Quantity",
    field: "quantity",
    editable: "never",
    cellStyle: {
      textAlign: "center",
      fontSize: "13px",
    },
    headerStyle: {
      textAlign: "center",
      fontSize: "13px",
    },
  },
  {
    title: "Request Type",
    field: "requestType",
    editable: "never",
    cellStyle: {
      textAlign: "center",
      fontSize: "13px",
    },
    headerStyle: {
      textAlign: "center",
      fontSize: "13px",
    },
  },
  {
    title: "Date",
    field: "scheduledDate",
    editable: "never",
    cellStyle: {
      textAlign: "center",
      fontSize: "13px",
    },
    headerStyle: {
      textAlign: "center",
      fontSize: "13px",
    },
  },
  {
    title: "Time",
    field: "scheduledTime",
    editable: "never",
    cellStyle: {
      textAlign: "center",
      fontSize: "13px",
    },
    headerStyle: {
      textAlign: "center",
      fontSize: "13px",
    },
  },
];
