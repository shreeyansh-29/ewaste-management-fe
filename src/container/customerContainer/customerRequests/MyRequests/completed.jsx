/*
  @module completed
 */
/* eslint-disable indent */
import React, {useEffect, useState} from "react";
import MaterialTable from "material-table";
import Popup from "../../../../components/popUp/popUp";
import "../../customer.css";
import {
  ProfileIconBarStyle,
  ProfileIconStyle,
} from "../../../../components/styles";
import {FaUserCircle} from "react-icons/fa";
import SearchIcon from "@material-ui/icons/Search";
import {toast} from "react-toastify";
export const ProfileIcon = FaUserCircle;
import {useDispatch, connect} from "react-redux";
import {customerCompletedRequest} from "../../../../redux/action/customer/customerCompletedRequestAction/customerCompletedRequestAction";
import {isEmpty} from "lodash";
import {completedColumns} from "./completedColumns";
toast.configure();

function completed({res}) {
  const dispatch = useDispatch();
  const [isopen, setopen] = useState(false);
  const [detail, setdetail] = useState();

  /* 
    @function togglepop
    @detail update the value of isopen
    @return {void}
  */
  const togglepop = () => {
    setopen(!isopen);
  };
  const handledata = (res1) => {
    res1.map((obj) => {
      if (obj.scheduledDate !== null) {
        const date = obj.scheduledDate.split("T");
        obj.scheduledDate = date[0];
      }
    });
  };
  useEffect(() => {
    if (isEmpty(res?.data) !== true) {
      handledata(res?.data.data);
    }
  });
  useEffect(() => {
    dispatch(customerCompletedRequest());
  }, []);

  return (
    <>
      <div>
        <div>
          <div style={{padding: " 10px 30px"}}>
            <MaterialTable
              align="center"
              title=""
              columns={completedColumns}
              data={res?.data.data}
              icons={{
                Search: () => <SearchIcon style={{fill: "white"}} />,
              }}
              localization={{
                header: {
                  actions: "Actions",
                },
              }}
              actions={[
                {
                  icon: () => (
                    <>
                      <ProfileIconStyle onClick={togglepop} id="pop1">
                        <ProfileIconBarStyle>
                          <ProfileIcon></ProfileIcon>
                        </ProfileIconBarStyle>
                      </ProfileIconStyle>
                    </>
                  ),

                  onClick: (e, datas) => {
                    e.preventDefault();
                    setdetail(datas.collectorUid);
                  },
                },
              ]}
              options={{
                actionsColumnIndex: -1,
              }}
            />
          </div>
        </div>
        <div>
          {isopen && detail != null && (
            <Popup id="pop" handleClose={togglepop} contents={detail} />
          )}
        </div>
      </div>
    </>
  );
}

const mapStateToProps = (state) => {
  return {
    res: state.customerCompletedRequest,
  };
};

export default connect(mapStateToProps)(completed);
