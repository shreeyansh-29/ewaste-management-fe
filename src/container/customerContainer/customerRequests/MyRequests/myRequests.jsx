/*
  @module myRequests
*/
import React, {useState} from "react";
import Tabs from "@mui/material/Tabs";
import Tab from "@mui/material/Tab";
import Paper from "@mui/material/Paper";
import Typography from "@material-ui/core/Typography";
import Box from "@material-ui/core/Box";
import Pending from "./pending";
import Completed from "./completed";
import "../../customer.css";

const MyRequests = () => {
  const [value, setValue] = useState(0);
  const handleChange = (event, newValue) => {
    event.preventDefault();
    setValue(newValue);
  };
  const TabPanel = (props) => {
    const {children, index, ...other} = props;
    return (
      <div
        role="tabpanel"
        hidden={value !== index}
        id={`simple-tabpanel-${index}`}
        aria-labelledby={`simple-tab-${index}`}
        {...other}
      >
        {value === index && (
          <Box>
            <Typography>{children}</Typography>
          </Box>
        )}
      </div>
    );
  };
  return (
    <div className="myRequests">
      <h2 className="myRequests-h2"> My Requests </h2>
      <Paper square className="myRequests-Paper">
        <Tabs
          value={value}
          indicatorColor="primary"
          textColor="primary"
          onChange={handleChange}
          aria-label="disabled tabs example"
          centered
          variant="fullWidth"
          elevation={20}
        >
          <Tab label="Pending" />

          <Tab label="Completed" />
        </Tabs>
        <TabPanel value={value} index={0}>
          <Pending />
        </TabPanel>
        <TabPanel value={value} index={1}>
          <Completed />
        </TabPanel>
      </Paper>
    </div>
  );
};
export default MyRequests;
