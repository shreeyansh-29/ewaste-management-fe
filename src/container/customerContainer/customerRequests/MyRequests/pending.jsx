/*
  @module pending
*/
import React, {useEffect} from "react";
import MaterialTable from "material-table";
import "../../customer.css";
import {FaUserCircle} from "react-icons/fa";
import SearchIcon from "@material-ui/icons/Search";
import {useDispatch, connect} from "react-redux";
import {customerPendingRequest} from "../../../../redux/action/customer/customerPendingRequestAction/customerPendingRequestAction";
import {customerPendingDeclineRequest} from "../../../../redux/action/customer/customerPendingRequestAction/customerPendingDeclineAction";
import {pendingColumns} from "./pendingColumns";
export const ProfileIcon = FaUserCircle;

const Pending = ({res}) => {
  const dispatch = useDispatch();
  /* 
    @function handleDecline
    @params {e,datas} specifies the data of the request user wants to decline
    @detail dispatch deletePendingRequest from deletePendingAction
  */
  const handleDecline = async (e, datas) => {
    e.preventDefault();

    dispatch(customerPendingDeclineRequest(datas.orderUid));
    document.location.reload();
  };
  useEffect(() => {
    dispatch(customerPendingRequest());
  }, []);

  return (
    <div>
      <div>
        <div className="pending">
          <MaterialTable
            align="center"
            title=""
            columns={pendingColumns}
            data={res?.data.data}
            icons={{
              Search: () => <SearchIcon className="pendingBtn" />,
            }}
            localization={{
              header: {
                actions: "Actions",
              },
            }}
            actions={[
              {
                icon: () => <button className="bttn"> Decline </button>,
                onClick: (e, datas) => handleDecline(e, datas),
              },
            ]}
            options={{
              actionsColumnIndex: -1,
            }}
          />
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    res: state.customerPendingRequest,
  };
};

export default connect(mapStateToProps)(Pending);
