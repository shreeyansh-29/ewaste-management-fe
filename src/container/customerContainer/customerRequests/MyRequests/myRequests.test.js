/* eslint-disable no-undef */
import React from "react";
import Enzyme, {mount, shallow} from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import toJson from "enzyme-to-json";
import MyRequests from "./myRequests";
import {Tabs} from "@mui/material";

Enzyme.configure({adapter: new Adapter()});

describe("testing myRequests", () => {
  it("should render Requests", () => {
    const wrapper = shallow(<MyRequests />);
    expect.assertions(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("should test onChange", () => {
    const onChange = jest.fn();
    const event = {
      preventDefault() {},
    };
    const wrapper = shallow(<MyRequests handleChange={onChange} />);
    wrapper.find(Tabs).simulate("change", event);
    expect(onChange).toHaveBeenCalled;
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("should test TabPanel", () => {
    const name = "TabPanel";
    const props = {children: "", index: "", other: ""};
    const TabPanel = jest.fn();
    const wrapper = mount(
      <MyRequests {...props} TabPanel={TabPanel}>
        {name}
      </MyRequests>
    );
    wrapper.find(TabPanel).exists();
    expect(TabPanel).toHaveBeenCalled;
  });
});
