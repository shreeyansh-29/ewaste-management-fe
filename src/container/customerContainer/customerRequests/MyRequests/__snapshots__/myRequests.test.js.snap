// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing myRequests should render Requests 1`] = `
<div
  className="myRequests"
>
  <h2
    className="myRequests-h2"
  >
     My Requests 
  </h2>
  <ForwardRef(Paper)
    className="myRequests-Paper"
    square={true}
  >
    <ForwardRef(Tabs)
      aria-label="disabled tabs example"
      centered={true}
      elevation={20}
      indicatorColor="primary"
      onChange={[Function]}
      textColor="primary"
      value={0}
      variant="fullWidth"
    >
      <ForwardRef(Tab)
        label="Pending"
      />
      <ForwardRef(Tab)
        label="Completed"
      />
    </ForwardRef(Tabs)>
    <TabPanel
      index={0}
      value={0}
    >
      <Connect(Pending) />
    </TabPanel>
    <TabPanel
      index={1}
      value={0}
    >
      <Connect(completed) />
    </TabPanel>
  </ForwardRef(Paper)>
</div>
`;

exports[`testing myRequests should test onChange 1`] = `
<div
  className="myRequests"
>
  <h2
    className="myRequests-h2"
  >
     My Requests 
  </h2>
  <ForwardRef(Paper)
    className="myRequests-Paper"
    square={true}
  >
    <ForwardRef(Tabs)
      aria-label="disabled tabs example"
      centered={true}
      elevation={20}
      indicatorColor="primary"
      onChange={[Function]}
      textColor="primary"
      variant="fullWidth"
    >
      <ForwardRef(Tab)
        label="Pending"
      />
      <ForwardRef(Tab)
        label="Completed"
      />
    </ForwardRef(Tabs)>
    <TabPanel
      index={0}
    >
      <Connect(Pending) />
    </TabPanel>
    <TabPanel
      index={1}
    >
      <Connect(completed) />
    </TabPanel>
  </ForwardRef(Paper)>
</div>
`;
